src/test/java/seedu/address/ui/CompletableDeadlineCardTest.java

```diff
@@ -0,0 +1,84 @@
+package seedu.address.ui;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.commons.util.DateUtil.encodeDate;
+import static seedu.address.ui.testutil.GuiTestAssert.assertCardDisplaysCompletableDeadline;
+
+import org.junit.jupiter.api.Test;
+
+import guitests.guihandles.CompletableDeadlineCardHandle;
+import seedu.address.commons.exceptions.DateConversionException;
+import seedu.address.model.task.CompletableDeadline;
+import seedu.address.testutil.DeadlineBuilder;
+
+/**
+ * Contains tests for the {@code CompletableDeadlineCard}.
+ */
+public class CompletableDeadlineCardTest extends GuiUnitTest {
+
+    private static final boolean DONE = true;
+    private static final boolean NOT_DONE = false;
+
+    @Test
+    public void display_success() throws DateConversionException {
+        // deadline is done
+        CompletableDeadline deadlineIsDone = new DeadlineBuilder().withDescription("Display Test")
+                .withByDate(encodeDate("18-03-2021")).withIsDone(DONE).build();
+        CompletableDeadlineCard deadlineCard = new CompletableDeadlineCard(deadlineIsDone, 1);
+        uiPartExtension.setUiPart(deadlineCard);
+        assertCardDisplay(deadlineCard, deadlineIsDone, 1);
+
+        // deadline is not done
+        CompletableDeadline deadlineIsNotDone = new DeadlineBuilder().withDescription("Display Test")
+                .withByDate(encodeDate("18-03-2021")).withIsDone(NOT_DONE).build();
+        deadlineCard = new CompletableDeadlineCard(deadlineIsNotDone, 2);
+        uiPartExtension.setUiPart(deadlineCard);
+        assertCardDisplay(deadlineCard, deadlineIsNotDone, 2);
+    }
+
+    @Test
+    public void equals() {
+        CompletableDeadline deadline = new DeadlineBuilder().build();
+        CompletableDeadlineCard deadlineCard = new CompletableDeadlineCard(deadline, 0);
+
+        // same deadline, same index -> returns true
+        CompletableDeadlineCard copy = new CompletableDeadlineCard(deadline, 0);
+        assertTrue(deadlineCard.equals(copy));
+
+        // same object -> returns true
+        assertTrue(deadlineCard.equals(deadlineCard));
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self comparison of value with itself (SA_LOCAL_SELF_COMPARISON)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-comparison-of-value-with-itself-sa-local-self-comparison).
This method compares a local variable with itself, and may indicate a typo or a logic error. Make sure that you are comparing the right things.